### DISCUSSION TITLE
Disable Authentication

### DISCUSSION THREAD

**Discussion by apocaliss92** [ZoeyVid/NPMplus#1297] [Q&A] [answered]
Is there the possibility to disable the authentication? I have always found a limitation of npm to not use an authentication system

**Comment by Zoey2936** [comment]
What is your usecase for this?

**Comment by apocaliss92** [comment]
Thank you for the super quick response!

I have authentik as SSO for all my services and npm is currently the only one which does not provide any possibility to disable the GUI authentication. There is an hack provided by an user but I could not really figure out how to make it work

For me it would really be enough to just disable it or allow passing basic authentication from the outside

**Comment by Zoey2936** [comment]
can you give me a link to this patch (or "hack") you mentioned?

Currently, you could just copy the Bearer token out of your browsers developer tools to make it work, but the token will is not valid very long. (1 day: https://github.com/ZoeyVid/NPMplus/blob/844ab284749f9fab971e3020228bd051a0ef0f58/backend/internal/token.js#L22)

But for things like authentic, I could add an env option to disable authentication, but I'm personally against using SSO, since you will have one password everywhere and since NPMplus is security relevant I don't think that this is a good idea

**Comment by apocaliss92** [comment]
Is it only required to pass it as Authorization header to each request? I will give it a try, I can generate tokens on authentik

The hack is something similar for npm

I will give it a try, thanks a lot

**Comment by Zoey2936** [comment]
Yes you only need to send the token (at least I think that this will work), the token can be creating by sending mail+password to the api, see this line as an example: it is an api request adding a host and requesting the token:
https://github.com/ZoeyVid/NPMplus/blob/844ab284749f9fab971e3020228bd051a0ef0f58/rootfs/usr/local/bin/aio.sh#L5

But still not recommended by me, but should work, if authentic can make this api request to get the token and then pass the token with your request

**Comment by Zoey2936** [comment]
I will convert this to a discussion, if you get it working, please leave a guide here if someone wants to achieve the same, but again not recommended by me. If you have other questions just ask

**Comment by apocaliss92** [comment]
Thank you very much for your hints, this is more complicate as expected, if one day there will be the possibility to just disable the authentication or implement a "trusted proxies" mechanism, I will be super happy :)
I completely agree with you on security issues, but my case is an home lab setup

Should I create some ticket for this specific request?

**Reply by Zoey2936** [reply]
no I will just leave this discussions "unanswered" so I will see it and can work on it when I have time

**Reply by apocaliss92** [reply]
As a side note, basic auth will work as well

**Comment by niffelheimba** [comment]
Just wanna say I would also support this option; I think the point about SSO making one large vulnerability is null for the same reasons it was invented in the first place (stop users from using the same repeatable weak passwords, only grants access to user authorized apps in case of a rare breach, etc.). I would just leave it as a configurable value up for users who want it.

**Comment by Zoey2936** [comment, marked as answer]
oauth will be supported with the next release
